### PR TITLE
[MAIN] [STRATCONN-2841] Added event_id in page action of Pinterest Tag

### PR DESCRIPTION
**Pull request by AnkitSegment**
**What does this PR do?**
JIRA TICKET: https://segment.atlassian.net/browse/STRATCONN-2841

Create mapping for Pinterest event_id as segment messageId. 

1. When `mapMessageIdToEventId` is disabled/false

Screenshot 2024-05-30 at 11 57 30 AM

Screenshot 2024-05-30 at 11 56 12 AM

2. When `mapMessageIdToEventId` is enabled/true

Screenshot 2024-05-30 at 11 29 53 AM

Screenshot 2024-05-30 at 11 52 59 AM

**Are there breaking changes in this PR?**

**Testing**

- Testing completed successfully using stage 

**Any background context you want to provide?**

**Is there parity with the server-side/android/iOS integration components (if applicable)?**

**Does this require a new integration setting? If so, please explain how the new setting works**

**Links to helpful docs and other external resources**

### CODE REVIEW

**Review by harsh-joshi99 — APPROVED** [review]
LGTM ✨

File: integrations/pinterest-tag/lib/index.js [hunk 1]
```diff
@@ -63,15 +63,27 @@ Pinterest.prototype.identify = function(identify) {
 
 Pinterest.prototype.page = function(page) {
   // If we have a category, the use ViewCategory. Otherwise, use a normal PageVisit.
+  var pinterestPageProps = {
+    name: page.name() || ''
+  };
+
+  var eventKeys = ['event_id', 'eid', 'eventID'];
```
Comment (varadarajan-tw) [inline integrations/pinterest-tag/lib/index.js:70]: Why do we search for `eid`, `eventID` these specific attributes? How did we decide these three values?

Comment (varadarajan-tw) [inline integrations/pinterest-tag/lib/index.js:70]:
```suggestion
  const eventKeys = ['event_id', 'eid', 'eventID'];
```

File: integrations/pinterest-tag/lib/index.js [hunk 2]
```diff
@@ -63,15 +63,27 @@ Pinterest.prototype.identify = function(identify) {
 
 Pinterest.prototype.page = function(page) {
   // If we have a category, the use ViewCategory. Otherwise, use a normal PageVisit.
+  var pinterestPageProps = {
```
Comment (varadarajan-tw) [inline integrations/pinterest-tag/lib/index.js:66]: you can mutate const object properties
```suggestion
  const pinterestPageProps = {
```

File: integrations/pinterest-tag/lib/index.js [hunk 3]
```diff
@@ -63,15 +63,27 @@ Pinterest.prototype.identify = function(identify) {
 
 Pinterest.prototype.page = function(page) {
   // If we have a category, the use ViewCategory. Otherwise, use a normal PageVisit.
```
Comment (varadarajan-tw) [inline integrations/pinterest-tag/lib/index.js:65]: Is this comment more relevant to the `if else` section?

**Review by varadarajan-tw — COMMENTED** [review]
Looks good. Left a few questions.

Also - could you bump the package version in `package.json`?

File: integrations/pinterest-tag/lib/index.js (hunk 1 above)
Comment (AnkitSegment) [reply integrations/pinterest-tag/lib/index.js:70]: [Pinterest only allows event IDs in these three formats. ](https://help.pinterest.com/en/business/article/add-event-codes#:~:text=advertiser%20account%20currency.-,event_id,-eventId0001)

Comment (AnkitSegment) [reply integrations/pinterest-tag/lib/index.js:70]: In this repo, Eslint is not allowing let and const variable declarations.

Comment (varadarajan-tw) [reply integrations/pinterest-tag/lib/index.js:69]: Okay. So, do we expect customers to provide these values in properties? Is this consistent with how we can handle `track`?

Comment (AnkitSegment) [reply integrations/pinterest-tag/lib/index.js:69]: Yes, for `track` events, mappings are generated through the `generatePropertiesObject` function. Using forLoop, when the user adds any of these three event IDs, that value directly gets attached to the mappings.

**Review by varadarajan-tw — APPROVED** [review]
